scipy/stats/_continuous_distns.py

```diff
@@ -8703,6 +8703,15 @@ def _logsf(self, x, a, b):
             logsf[i] = np.log1p(-np.exp(self._logcdf(x[i], a[i], b[i])))
         return logsf
 
+    def _entropy(self, a, b):
+        A = _norm_cdf(a)
+        B = _norm_cdf(b)
+        Z = B - A
+        C = np.log(np.sqrt(2 * np.pi * np.e) * Z)
```

Comment:
```suggestion
        logZ = _log_gauss_mass(a, b)
        return (np.log(np.sqrt(2 * np.pi * np.e)) + logZ +
                (a * _norm_pdf(a) - b * _norm_pdf(b)) / (2 * np.exp(logZ)))
```

Reply:
I get slightly different reference values with the following snippet (for the last test case `-34.53877639491019`):

```
def truncnorm_mpmath_entropy(a, b):
    a = mp.mpf(a)
    b = mp.mpf(b)
    mp.dps = 500
    A = mp.ncdf(a)
    B = mp.ncdf(b)
    Z = B - A
    C = mp.log(mp.sqrt(2 * mp.pi * mp.e)) + mp.log(Z)
    D = (a * mp.npdf(a) - b * mp.npdf(b))/(2 * Z)
    return float(C + D)
```

The crucial point is the conversion of the inputs to multiprecision floatsvia `mp.mpf()`. I think I had pointed to another post here in the first entropy related PR @MatteoRaso , sorry for missing it here before. The reference values for the tests need to be recomputed unfortunately.

Reply:
The tricky thing is when both `a` and `b` are both deep in the same tail. The tests in `test_extreme_entropy` right now are essentially the same as the half-normal and regular normal distribution because one of the truncation points is either near 0 or deep in the opposite tail. It looks like we may only see the shortcomings of this implementation in that case.

Reply:
> This doesn't happen when using the current formula.

This also tends to happen when you produce the reference values using the PR implementation.

```python3
import mpmath as mp
mp.dps = 50
def entropy_trun(a, b):
    def cdf(x):
        return 0.5 * (1 + mp.erf(x / mp.sqrt(2)))

    Z = cdf(b) - cdf(a)

    def pdf_standard_norm(x):
        return (mp.exp(-0.5 * x ** 2) / mp.sqrt(2 * mp.pi))
    def pdf(x):
        return pdf_standard_norm(x) / Z

    return -mp.quad(lambda t: pdf(t) * mp.log(pdf(t)), [a, b])

from scipy.stats import truncnorm
truncnorm.entropy(a, b)  # -34.8181418495907, the reference value in the test
entropy_trun(a, b)  # -34.538776394910184650320109336919011124273984815566988
```